src/WebJobs.Script/Description/DotNet/DependencyHelper.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.DependencyModel;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public static class DependencyHelper
+    {
+        private static readonly Lazy<Dictionary<string, string[]>> _ridGraph = new Lazy<Dictionary<string, string[]>>(BuildRuntimesGraph);
+
+        private static Dictionary<string, string[]> BuildRuntimesGraph()
+        {
+            var ridGraph = new Dictionary<string, string[]>();
+            string runtimesJson = GetRuntimesGraphJson();
+            var runtimes = (JObject)JObject.Parse(runtimesJson)["runtimes"];
+
+            foreach (var runtime in runtimes)
+            {
+                string[] imports = ((JObject)runtime.Value)["#import"]
+                    ?.Values<string>()
+                    .ToArray();
+
+                ridGraph.Add(runtime.Key, imports);
+            }
+
+            return ridGraph;
+        }
+
+        public static IEnumerable<string> GetFallbacks(Dictionary<string, string[]> ridGraph, string rid)
+        {
+            var fallbacks = new HashSet<string>();
+            var queue = new Queue<string>(ridGraph[rid]);
+
+            while (queue.Count > 0)
+            {
+                var currentRid = queue.Dequeue();
+
+                if (fallbacks.Contains(currentRid))
+                {
+                    continue;
+                }
+
+                fallbacks.Add(currentRid);
+
+                foreach (var fallbackRid in ridGraph[currentRid])
+                {
+                    if (!fallbacks.Contains(fallbackRid))
+                    {
+                        queue.Enqueue(fallbackRid);
+                    }
+                }
+            }
+
+            return fallbacks;
+        }
+
+        private static string GetRuntimeAssembliesJson()
+        {
+            return GetResourceFileContents("runtimeassemblies.json");
+        }
+
+        private static string GetRuntimesGraphJson()
+        {
+            return GetResourceFileContents("runtimes.json");
+        }
+
+        private static string GetResourceFileContents(string fileName)
+        {
+            var assembly = typeof(DependencyHelper).Assembly;
+            using (Stream resource = assembly.GetManifestResourceStream($"{assembly.GetName().Name}.{fileName}"))
```

Comment:
Consider adding a helper for this, since we have this same resource load code in 3 other places currently

Reply:
That was a helper :) Consolidated the other calls to use this, but will introduce a helper to be used by others as well.